### PR TITLE
Add FunctionalIntegrationTestCase class and extensions, plus example articles_controller_integration_test

### PR DESCRIPTION
**Pull request by nimmolo**
This PR aims to create a parallel new class for our functional tests that inherits from `ActionDispatch::IntegrationTest` instead of `ActionController::TestCase`. 

It also includes an example integration-style controller test, for Articles, which attempts to exactly duplicate the existing controller test, in parallel. It's called `articles_controller_integration_test.rb`

Here are the new files added:
```
- test
-- functional_integration_test_case.rb
-- controller_integration_extensions.rb
-- controllers
--- articles_controller_integration_test.rb
```
Revised files
```
- app
-- controllers
--- application_controller.rb (adding a line to include ObjectLinkHelper at the top)
-- helpers
--- object_link_helper.rb (adding the new object link methods)
- test
-- test_helper.rb (requiring the new files above in the list of test cases and extensions)
```
So far, so good. (The integration-controller test runs in 8 seconds, just like the controller test it's based on!)
All tests in the new `articles_controller_integration_test` are green in this branch, except one assertion in `test_update`.
Here's that part of the revised method. 
```
    # Prove removing title provokes warning
    params[:article][:title] = ""
    patch(article_path(params))
    # NOTE: The flash text seems to give all possible messages at once.
    # <p>Successfully updated article #788338338.</p><p>No changes made.</p><p>Title Required</p>
    assert_flash_text(:article_title_required.l)
    assert_equal(edit_article_path(params), path)
    assert_form_action(action: :update) # "edit" form
```
(Note that with CRUD resources/paths, you PUT or PATCH to the article_path(params) to update the article, as shown in rake routes for articles currently.)
Here's the error message:
```
        Flash text incorrect.
        --- expected
        +++ actual
        @@ -1 +1 @@
        -"<p>Title Required</p>"
        +"<p>Successfully updated article #788338338.</p><p>No changes made.</p><p>Title Required</p>"
        test/flash_extensions.rb:84:in `assert_flash_text'
        test/controllers/articles_controller_test.rb:220:in `test_update'
```
I wonder if maybe the integration test is reporting the actual response accurately?

Maybe someone who is an actual admin can try removing the title of an article, and see what flash response you get in the wild.

### CODE REVIEW

**Comment by nimmolo** [comment]
Copied from email - Joe has tested the edit-article form in the wild and he does NOT get multiple flash messages.
Hmm.

**Comment by JoeCohen** [comment]
Nimmo: I probably won't have time to dig into this until Fri.
I did notice that the gems are out of date:
```
vagrant@ubuntu-focal /vagrant/mushroom-observer (nimmo-controller-integration-tests-09-20)
$ rails lang:update
Could not find proper version of railties (5.2.4.3) in any of the sources
```
master is on railties 5.2.6

**Comment by nimmolo** [comment]
Thats weird. I forked this branch straight off Master yesterday. I did not edit railties version afaik. 

Feel free to edit the gemfile and bundle install again!
